pyiron/dft/waves/electronic.py

```diff
@@ -446,14 +446,17 @@ def to_hdf_new(self, hdf, group_name="band"):
                 if self.resolved_densities is not None:
                     h_dos["resolved_densities"] = self.resolved_densities
 
-    def from_hdf_new(self, hdf, group_name="bands"):
+    def from_hdf(self, hdf, group_name="bands"):
         """
         Retrieve the object from the hdf5 file
 
         Args:
             hdf: Path to the hdf5 file/group in the file
             group_name: Name of the group under which the attributes are stored
         """
+        if 'dos' not in hdf[group_name].list_groups():
+            self.from_hdf_old(hdf=hdf, group_name=group_name)
+            return
```

Comment:
Is there as specific reason to use `if` and then `return` instead of `if else`, to me `if else` is more readable dude to the indentation.

Reply:
Done: https://github.com/pyiron/pyiron/pull/61/commits/8a533962e8915c82299713d76bc4b1631afd186d and https://github.com/pyiron/pyiron/pull/61/commits/abdeb66c3d632bd2b2b0771be59a4134ab763ef3